src/com/directi/training/codesmells/refactored/chess/ChessBoard.java

```diff
@@ -125,8 +125,6 @@ public boolean isValidMove(Position from, Position to)
     // by extracting out code for knight in a separate method.
     private boolean hasNoPieceInPath(Position from, Position to)
     {
-        if (getPiece(from) instanceof Knight)
-            return hasNoPieceInPathOfKnight(from, to);
```

Comment:
We would still need to return true if it is a Knight since rest of the conditions won't apply.